Problem3.rb

```diff
@@ -0,0 +1,48 @@
+require 'rspec'
+
+def swap(array, i, j)
+
+  temp = array[i]
+  array[i] = array[j]
+  array[j] = temp
+
+end
+
+def contains_duplicates?(array)
+
+  min = 1
+
+  array.each_index do |i|
+
+    temp = array[i]
+    j = temp - min
+
+    while i != j do
+
+      if temp == array[j]
+        return true

```

Comment:
Please keep a single return per function. Ruby, to help with this, returns the result of the last line in the method, without needing to specifically write `return`. Ex:

```
def sum(a,b)
  a + b
end

sum(1,3) # returns 4
```